src/category_theory/adjunction/basic.lean

```diff
@@ -204,16 +204,87 @@ def id : 𝟭 C ⊣ 𝟭 C :=
   unit := 𝟙 _,
   counit := 𝟙 _ }
 
+/-- If F and G are naturally isomorphic functors, establish an equivalence of hom-sets. -/
+def equiv_homset_left_of_nat_iso
+  {F G : C ⥤ D} (iso : F ≅ G) {X : C} {Y : D} :
+  (F.obj X ⟶ Y) ≃ (G.obj X ⟶ Y) :=
+{ to_fun := λ f, iso.inv.app _ ≫ f,
+  inv_fun := λ g, iso.hom.app _ ≫ g,
+  left_inv := λ f, by simp,
+  right_inv := λ g, by simp }
+
+@[simp]
```

Comment:
You should be able to generate these two lemmas automatically by putting `@[simps]` on `equiv_homset_left_of_nat_iso`.

Reply:
I did try this, it doesn't do the job: it generates `equiv_homset_left_of_nat_iso_to_fun` with type
```lean
(equiv_homset_left_of_nat_iso iso).to_fun f = ...
```
which isn't good enough, it's the function coercion that's necessary here. (I originally tried simps, then reducible and removed that based on Scott's suggestions above)